### ISSUE TITLE
addic7ed: AttributeError: 'NoneType' object has no attribute 'group'

### ISSUE THREAD

**Issue by labrys** [Diaoul/subliminal#654]
Subliminal 2.0rc1
```
2016-04-27 22:17:04 FINDSUBTITLES :: [6ac2d37] Unexpected error in provider 'addic7ed'
Traceback (most recent call last):
  File "build\bdist.win-amd64\egg\subliminal\core.py", line 118, in list_subtitles_provider
    return self[provider].list_subtitles(video, provider_languages)
  File "build\bdist.win-amd64\egg\subliminal\providers\addic7ed.py", line 269, in list_subtitles
    return [s for s in self.query(video.series, video.season, video.year)
  File "build\bdist.win-amd64\egg\subliminal\providers\addic7ed.py", line 239, in query
    series = match.group('series')
AttributeError: 'NoneType' object has no attribute 'group'
```

**Comment by Diaoul** [comment]
See https://github.com/Diaoul/subliminal/blob/master/CONTRIBUTING.md

**Comment by labrys** [comment]
@Diaoul Sorry for the admittedly poor issue request.  This is being run from within SickRage and I attempted to get more information for it, however when it happens, the log files have usually already rolled over.  I kept intending to get back to this and try to provide additional information.

**Comment by fdbgbgsf** [comment]
Same issue here. No subtitle downloads for addic7ed at all. No matter the search method (auto, full, per episode, …) The log is the exact same as labrys'

Also using SickRage, If there's any additional info you like me to provide, let me know.

- Branch: master
- Commit: f970b5fd5e222469059e71c9b0d6a18348d2770e
- Version: 2016.05.12-1
- Database Version: 43.1
- Python Version: 2.7.11 (default, Dec 17 2015, 15:58:38) [GCC 4.7.3 20130102 (prerelease)]
- OS: Linux-3.10.77-x86_64-with-glibc2.2.5
- Synology 1813+
- DSM 6.0-7321 Update 6

**Comment by labrys** [comment]
I have had this once since, but again logs had rolled over so no useful information to add.

**Comment by Malka** [comment]
Just had the very same issue. Here are the logs:

```
root@legarage /h/P/t/T/Season 9# subliminal --debug --addic7ed ***** ***** download -p addic7ed  -l fr -l en The\ Office\ \(US\)\ -\ S09E01\ -\ New\ Guys\ DVD.mkv  2> debug.log
Collecting videos  [####################################]  100%                                             
1 video collected / 0 video ignored / 0 error
Downloading subtitles  [####################################]  100%                                             
Downloaded 0 subtitle
root@legarage /h/P/t/T/Season 9# cat debug.log 
DEBUG:subliminal.cli:Collecting path The Office (US) - S09E01 - New Guys DVD.mkv
INFO:subliminal.video:Scanning video 'The Office (US) - S09E01 - New Guys DVD.mkv' in ''
DEBUG:subliminal.video:Size is 227137028
DEBUG:subliminal.video:Computed hashes {'opensubtitles': 'a327a927657d35d8', 'thesubdb': '19122501882738402fe64de25ff5377e', 'napiprojekt': '990f8d489876154f31760a5037b2a173'}
DEBUG:subliminal.video:Found subtitles {'The Office (US) - S09E01 - New Guys DVD.en.srt': <Language [en]>}
DEBUG:subliminal.video:Found video_codec h264 with enzyme
DEBUG:subliminal.video:Found audio_codec AAC with enzyme
DEBUG:subliminal.video:MKV has no subtitle track
INFO:subliminal.api:Listing subtitles with provider 'addic7ed' and languages set([<Language [fr]>])
INFO:subliminal.api:Initializing provider addic7ed
INFO:subliminal.providers.addic7ed:Logging in
DEBUG:subliminal.providers.addic7ed:Logged in
DEBUG:subliminal.providers.addic7ed:Getting show id
INFO:subliminal.providers.addic7ed:Getting the page of show id 65, season 9
ERROR:subliminal.api:Unexpected error in provider 'addic7ed', discarding it
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/subliminal/api.py", line 116, in list_subtitles
    provider_subtitles = self[name].list_subtitles(video, provider_languages)
  File "/usr/local/lib/python2.7/site-packages/subliminal/providers/addic7ed.py", line 259, in list_subtitles
    return [s for s in self.query(video.series, video.season, video.year)
  File "/usr/local/lib/python2.7/site-packages/subliminal/providers/addic7ed.py", line 229, in query
    series = match.group('series')
AttributeError: 'NoneType' object has no attribute 'group'
DEBUG:subliminal.api:Terminating initialized providers
INFO:subliminal.api:Terminating provider addic7ed
INFO:subliminal.providers.addic7ed:Logging out
DEBUG:subliminal.providers.addic7ed:Logged out

```

**Comment by Diaoul** [comment]
Fixing this would require adding country support in Episode class and enable support in refiners.

**Comment by labrys** [comment]
@Diaoul would there be unwanted side effects to `series = match.group('series') if match else ''`?

**Comment by Diaoul** [comment]
Yes, the side effect that I won't be able to improve the parsing. At least
this way it gets reported.
Maybe some error logging would be better but... (see above)

**Comment by labrys** [comment]
Good reason :grinning:

**Comment by nudep** [comment]
after receiving same error
INFO:subliminal.api:Listing subtitles with provider 'addic7ed' and languages set([<Language [en]>])
INFO:subliminal.api:Initializing provider addic7ed
INFO:subliminal.providers.addic7ed:Logging in
DEBUG:subliminal.providers.addic7ed:Logged in
DEBUG:subliminal.providers.addic7ed:Getting show id
INFO:subliminal.providers.addic7ed:Getting the page of show id 5814, season 1
ERROR:subliminal.api:Unexpected error in provider 'addic7ed', discarding it
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/subliminal/api.py", line 116, in list_subtitles
    provider_subtitles = self[name].list_subtitles(video, provider_languages)
  File "/usr/local/lib/python2.7/dist-packages/subliminal/providers/addic7ed.py", line 259, in list_subtitles
    return [s for s in self.query(video.series, video.season, video.year)
  File "/usr/local/lib/python2.7/dist-packages/subliminal/providers/addic7ed.py", line 229, in query
    series = match.group('series')
AttributeError: 'NoneType' object has no attribute 'group'

tried remaking same steps inside addic7ed.py from python command line
in this specific case problem was with the name of the show containing a comma "," inside
after adding the comma (?P<series>[ \w\'.,:]+) instead of (?P<series>[ \w\'.:]+)
the series_year_re regular expression returns the match

it seems that Malka's problem with The Office (US) can be resolved in a similar way
adding opening and closing parenthesis to the series part of the series_year_re
(?P<series>[ \w\'.,\(\):]+) 
returns the match.group(0) u'The Office (US)'
instead of AttributeError: 'NoneType' object has no attribute 'group'

@Diaoul Sorry for not sending this via pull request. I (almost) know how to use git but... well... how to say this. I still have to gather some confidence.

Hope this helps

**Comment by nudep** [comment]
(?P\<series\>[ \w\'.,:]+)  for only the comma
(?P\<series\>[ \w\'.,\\(\\):]+)  for comma and parenthesis
sorry forgot to escaped and double escape characters for the correct html visualization

Cheers

**Comment by donbi** [comment]
Thanks @nudep, this works for me.

**Comment by Diaoul** [comment]
Fixed with ab2523c4af206caba4e61dd9cffda356753b98ae
